src/Hudl.Mjolnir/Command/BulkheadInvoker.cs

```diff
@@ -123,7 +125,7 @@ public TResult ExecuteWithBulkhead<TResult>(SyncCommand<TResult> command, Cancel
             }
             finally
             {
-                bulkhead.Release();
+                if (!onlyMetrics) bulkhead.Release();
 
                 _metricEvents.LeaveBulkhead(bulkhead.Name, command.Name);
```

Comment:
One more question. Do we want the LeveBulkhead metrics for those calls which would have tripped bulkhead if only metrics was not set?

Reply:
I think we do, as it will give the full picture. In reality the call has gone through the bulkhead successfully, because it's disabled. But I could see it skewing metrics if they were being collected in a different way to logs, so I could be swayed to not make the Enter and Leave calls in this case.

Reply:
I guess, as long as it's not confusing and clear what we want to achieve here I'm ok with anything. Just wanted to highlight how it works now, and make sure this is the intention.